benchexec/test_runexecutor.py

```diff
@@ -1199,6 +1200,135 @@ def test_uptime_without_lxcfs(self):
             uptime, 10, f"Uptime {uptime}s unexpectedly low in container"
         )
 
+    def test_fuse_overlay(self):
+        if not container.get_fuse_overlayfs_executable():
+            self.skipTest("fuse-overlayfs not available")
+        with tempfile.TemporaryDirectory(prefix="BenchExec_test_") as temp_dir:
+            test_file_path = os.path.join(temp_dir, "test_file")
+            with open(test_file_path, "wb") as test_file:
+                test_file.write(b"TEST_TOKEN")
+
+            self.setUp(
+                dir_modes={
+                    "/": containerexecutor.DIR_READ_ONLY,
+                    "/home": containerexecutor.DIR_HIDDEN,
+                    "/tmp": containerexecutor.DIR_HIDDEN,
+                    temp_dir: containerexecutor.DIR_OVERLAY,
+                },
+            )
+            result, output = self.execute_run(
+                "/bin/sh",
+                "-c",
+                f"if [ $({self.cat} {test_file_path}) != TEST_TOKEN ]; then exit 1; fi; \
+                {self.echo} TOKEN_CHANGED >{test_file_path}",
+            )
+            self.check_result_keys(result, "returnvalue")
+            self.check_exitcode(result, 0, "exit code of inner runexec is not zero")
+            self.assertTrue(
+                os.path.exists(test_file_path),
+                f"File '{test_file_path}' removed, output was:\n" + "\n".join(output),
+            )
+            with open(test_file_path, "rb") as test_file:
+                test_token = test_file.read()
+            self.assertEqual(
+                test_token.strip(),
+                b"TEST_TOKEN",
+                f"File '{test_file_path}' content is incorrect. Expected 'TEST_TOKEN', but got:\n{test_token}",
+            )
+
+    def test_triple_nested_runexec(self):
+        if not container.get_fuse_overlayfs_executable():
+            self.skipTest("missing fuse-overlayfs")
+
+        # Check if COV_CORE_SOURCE environment variable is set and remove it.
+        # This is necessary because the coverage tool will not work in the nested runexec.
+        restore_env = False
+        if "COV_CORE_SOURCE" in os.environ:
+            restore_env = True
+            del os.environ["COV_CORE_SOURCE"]
```

Comment:
I would prefer if we do not have to hard-code the value of this variable, let's store it:
```suggestion
        coverage_env_var = os.environ.pop("COV_CORE_SOURCE", None)
```
Then later on we can do
```python
        if coverage_env_var is not None:
            os.environ["COV_CORE_SOURCE"] = coverage_env_var
```